sale_category_pricelist/models/product_pricelist_item.py

```diff
@@ -0,0 +1,123 @@
+            if item.sale_categ_id and item.applied_on == "2_0_product_sale_category":
+                item.name = _(f"Sale Category: {item.sale_categ_id.display_name}")
+        return res
+
+    @api.constrains("sale_categ_id")
+    def _check_product_consistency(self):
+        res = super()._check_product_consistency()
+        for item in self:
+            if (
+                item.applied_on == "2_0_product_sale_category"
+                and not item.sale_categ_id
+            ):
+                raise ValidationError(
+                    _(
+                        "Please specify the sale category for which "
+                        "this rule should be applied"
+                    )
+                )
+        return res
+
+    def create(self, vals_list):
+        for values in vals_list:
+            if values.get("applied_on", False):
+                # Ensure item consistency for later searches.
+                applied_on = values["applied_on"]
+                if applied_on == "3_global":
+                    values.update(
+                        dict(
+                            product_id=None,
+                            product_tmpl_id=None,
+                            sale_categ_id=None,
+                            categ_id=None,
+                        )
+                    )
+                elif applied_on == "2_product_category":
+                    values.update(
+                        dict(product_id=None, product_tmpl_id=None, sale_categ_id=None)
+                    )
+                elif applied_on == "2_0_product_sale_category":
+                    values.update(dict(product_id=None, product_tmpl_id=None))
+                elif applied_on == "1_product":
+                    values.update(dict(product_id=None, categ_id=None))
+                elif applied_on == "0_product_variant":
+                    values.update(dict(categ_id=None))
+        return super().create(vals_list)
+
+    def write(self, values):
+        if values.get("applied_on", False):
+            # Ensure item consistency for later searches.
+            applied_on = values["applied_on"]
+            if applied_on == "3_global":
+                values.update(
+                    dict(
+                        product_id=None,
+                        product_tmpl_id=None,
+                        categ_id=None,
+                        sale_categ_id=None,
+                    )
+                )
+            elif applied_on == "2_product_category":
+                values.update(
+                    dict(product_id=None, product_tmpl_id=None, sale_categ_id=None)
+                )
+            elif applied_on == "2_0_product_sale_category":
+                values.update(
+                    dict(product_id=None, product_tmpl_id=None, categ_id=None)
+                )
+            elif applied_on == "1_product":
+                values.update(dict(product_id=None, categ_id=None, sale_categ_id=None))
```

Comment:
You change sale_categ_id here but not in create method. Is that correct?

Reply:
No, changed